examples/citi_bike/online_lp/citi_bike_lp.py

```diff
@@ -0,0 +1,187 @@
+                    lowBound=0,
+                    upBound=round(self._safety_inventory_limit * self._station_capacity[station]),
+                    cat="Integer"
+                )
+                self._fulfillment[tick][station] = LpVariable(
+                    name=f"T{tick}_S{station}_Fulfillment", lowBound=0, cat="Integer"
+                )
+
+                # For intermediate variables.
+                self._transfer_from[tick][station] = LpVariable(
+                    name=f"T{tick}_TransferFrom{station}", lowBound=0, cat="Integer"
+                )
+                self._transfer_to[tick][station] = LpVariable(
+                    name=f"T{tick}_TransferTo{station}", lowBound=0, cat="Integer"
+                )
+
+                for neighbor_idx in range(self._num_neighbor):
+                    self._transfer[tick][station][neighbor_idx] = LpVariable(
+                        name=f"T{tick}_Transfer_from{station}_to{neighbor_idx}th", lowBound=0, cat="Integer"
+                    )
+
+        # Initialize the current inventory.
+        for station in range(self._num_station):
+            self._inventory[0][station] = init_inventory[station]
+
+    def _add_constraints(self, problem: LpProblem, demand: np.ndarray, supply: np.ndarray):
+        for tick in range(self._num_tick):
+            for station in range(self._num_station):
+                problem += (
+                    self._fulfillment[tick][station] <= demand[tick, station]
+                ), f"Fulfillment_Limit_T{tick}_S{station}"
+                # For intermediate variables.
+                problem += (
+                    self._transfer_from[tick][station] == lpSum(
+                        self._transfer[tick][station][neighbor_idx] for neighbor_idx in range(self._num_neighbor)
+                    )
+                ), f"TotalTransferFrom_T{tick}_S{station}"
+                problem += (
+                    self._transfer_to[tick][station] == lpSum(
+                        self._transfer[tick][neighbor][self._station_neighbor_list[neighbor].index(station)]
+                        for neighbor in range(self._num_station)
+                        if station in self._station_neighbor_list[neighbor][:self._num_neighbor]
+                    )
+                ), f"TotalTransferTo_T{tick}_S{station}"
+
+        for tick in range(1, self._num_tick):
+            for station in range(self._num_station):
+                problem += (
+                    self._inventory[tick][station] == (
+                        self._inventory[tick - 1][station]
+                        + supply[tick - 1, station] - self._fulfillment[tick - 1][station]
+                        + self._transfer_to[tick - 1][station] - self._transfer_from[tick - 1][station]
+                    )
+                ), f"Inventory_T{tick}_S{station}"
+                problem += (
+                    self._safety_inventory[tick][station] <= (
```

Comment:
So the safety inventory is a dynamic low watermark, which is used to measure minimal inventory for no extra-supply (transfer to) case, and it can be zero?
`We may need an offline discussion for it.`

Reply:
It is a reward for planning. The minimum reward is 0, the maximum reward is limited by the ratio in the config.